sources/editor/Stride.Assets.Presentation/AssetEditors/EntityHierarchyEditor/Game/EditorGameComponentGizmoService.cs

```diff
@@ -314,8 +319,15 @@ private void CreateGizmoEntity(Entity entity, EntityComponent component, Type gi
                 return;
 
             // initialize the gizmo
-            var gizmo = (EntityGizmo)Activator.CreateInstance(gizmoType, component);
-            gizmo.InitializeContentEntity(entity);
+            IEntityGizmo gizmo;
+            try
+            {
+                gizmo = (IEntityGizmo)Activator.CreateInstance(gizmoType, component);
+            }
+            catch (MissingMethodException e)
+            {
+                throw new MissingMethodException($"{nameof(IEntityGizmo)} '{gizmoType}' must have a constructor with exactly one parameter of type '{component.GetType()}'", e);
```

Comment:
Was the default exception message not satisfying?

Reply:
This would output `System.MissingMethodException: Constructor on type 'Gizmo' not found.`, it is a bit misleading since you may very well have a constructor on that type, but we require a constructor with exactly one argument of a specific type.
Just making things clearer for our users.